packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,299 @@
+                )),
+                _ => Err(BuildRequirePathError::FileNameNotReal),
+            },
+            _ => {
+                let config =
+                    relay_compiler::config::Config::search(&std::env::current_dir().unwrap())
+                        .unwrap();
+
+                let categorizer = FileCategorizer::from_config(&config);
+
+                match &self.file_name {
+                    FileName::Real(real_file_name) => {
+                        // Make sure we have a path which is relative to the config.
+                        // Otherwise, categorize won't be able to recognize that
+                        // the absolute source path is a child of a source set.
+                        let diffed_path = diff_paths(real_file_name, &config.root_dir).unwrap();
+
+                        let group = categorizer.categorize(diffed_path.as_path());
+
+                        match group {
+                            Ok(group) => match group {
+                                FileGroup::Source { source_set } => match source_set {
+                                    SourceSet::SourceSetName(source_set_name) => {
+                                        let project_config: Option<&ProjectConfig> =
+                                            config.projects.get(&source_set_name);
+
+                                        match project_config {
+                                            None => Err(BuildRequirePathError::ProjectNotFoundForSourceSet { source_set_name }),
+                                            Some(project_config) => {
+                                                Ok(path_for_artifact(&config.root_dir,real_file_name, &project_config, operation_name))
+                                            }
+                                        }
+                                    }
+                                    SourceSet::SourceSetNames(source_set_names) => {
+                                        Err(BuildRequirePathError::MultipleSourceSetsFound {
+                                            source_set_names,
+                                            path: real_file_name,
+                                        })
+                                    }
+                                },
+                                _ => Err(BuildRequirePathError::FileNotASourceFile),
+                            },
+                            Err(err) => Err(BuildRequirePathError::CouldNotCategorize {
+                                err,
+                                path: real_file_name.display().to_string(),
+                            }),
+                        }
+                    }
+                    _ => Err(BuildRequirePathError::FileNameNotReal),
+                }
+            }
+        }
+    }
+
+    fn build_call_expr_from_tpl(&mut self, tpl: &TaggedTpl) -> Option<Expr> {
+        if let Expr::Ident(ident) = &*tpl.tag {
+            if &*ident.sym != "graphql" {
```

Comment:
What should happen if `graphql` is a declared variable?
Also, should `graphql` be imported from a specific module?

Reply:
Does not need to be imported from a specific module. @kdy1 Is there a way to make sure they didn't try to redeclare graphql? If it's easy we may as well add it. Should I just look for all VarDeclarators?

Reply:
It looks like `babel-plugin-relay` itself isn't doing anything special: https://github.com/facebook/relay/blob/ee374c8e2f0c4d75c64e3f76ac61032f52836760/packages/babel-plugin-relay/getValidGraphQLTag.js#L25

Reply:
We can check for variable declarators, but it's fine anyway as people using the babel relay plugin cannot declare a variable named `graphql`.